src/main/java/org/sonar/plugins/csharp/CSharpSensor.java

```diff
@@ -104,14 +104,41 @@ void executeInternal(SensorContext context) {
     String roslynReportPath = settings.getString(ROSLYN_REPORT_PATH_PROPERTY_KEY);
     boolean hasRoslynReportPath = roslynReportPath != null;
 
-    analyze(!hasRoslynReportPath, context);
-    importResults(context);
+    String analyzerWorkDirPath = settings.getString(ANALYZER_WORKDIR_PATH_PROPERTY_KEY);
+    boolean requiresAnalyzerScannerExecution = requiresAnalyzerScannerExecution(analyzerWorkDirPath);
+
+    LOG.info("SonarAnalyzer.Scanner needs to be executed: " + requiresAnalyzerScannerExecution);
+
+    if (requiresAnalyzerScannerExecution) {
+      analyze(!hasRoslynReportPath, context);
+    }
+
+    File workDirectory = requiresAnalyzerScannerExecution
+      ? toolOutput(context.fileSystem())
+      : analyzerOutputDir(analyzerWorkDirPath);
+
+    LOG.info("Importing analysis results from " + workDirectory.getAbsolutePath());
+    importResults(context, workDirectory);
 
     if (hasRoslynReportPath) {
+      LOG.info("Importing Roslyn report");
       importRoslynReport(roslynReportPath, context);
     }
   }
 
+  private boolean requiresAnalyzerScannerExecution(String analyzerWorkDirPath) {
+    File dir = analyzerOutputDir(analyzerWorkDirPath);
+    if (!dir.exists()){
+      LOG.info("Analyzer working directory does not exist");
+      return true;
+    }
+
+    File[] files = dir.listFiles((d, name) -> name.toLowerCase().endsWith(".pb"));
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Define the locale to be used in this String operation. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1449)